inc/saisegmentroute.h

```diff
@@ -0,0 +1,168 @@
+/**
+ * Copyright (c) 2017 Microsoft Open Technologies, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *    not use this file except in compliance with the License. You may obtain
+ *    a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR
+ *    CONDITIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT
+ *    LIMITATION ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS
+ *    FOR A PARTICULAR PURPOSE, MERCHANTABILITY OR NON-INFRINGEMENT.
+ *
+ *    See the Apache Version 2.0 License for specific language governing
+ *    permissions and limitations under the License.
+ *
+ *    Microsoft would like to thank the following companies for their review and
+ *    assistance with these files: Intel Corporation, Mellanox Technologies Ltd,
+ *    Dell Products, L.P., Facebook, Inc
+ *
+ * @file    saisegmentroute.h
+ *
+ * @brief   This module defines SAI Segment Route Entry interface
+ */
+
+#if !defined (__SAISEGMENTROUTE_H_)
+#define __SAISEGMENTROUTE_H_
+
+#include <saitypes.h>
+
+/**
+ * @defgroup SAISEGMENTROUTE SAI - Segment Route specific API definitions
+ *
+ * @{
+ */
+
+/**
+ * @brief Attribute data for SAI Segment Route Entry
+ */
+typedef enum _sai_segmentroute_entry_attr_t
+{
+    /**
+     * @brief Start of attributes
+     */
+    SAI_SEGMENTROUTE_ENTRY_ATTR_START = 0x00000000,
+
+    /**
+     * @brief Number of Segments supported on device for origination
+     *
+     * @type sai_uint8_t
+     * @flags READ_ONLY
+     */
+    SAI_SEGMENTROUTE_ENTRY_ATTR_SUPPORTED_NUM_SEGMENTS = SAI_SEGMENTROUTE_ENTRY_ATTR_START,
+
+    /**
+     * @brief List of TLV types supported on device for origination
+     *
+     * @type sai_s32_list_t sai_tlv_type_t
+     * @flags READ_ONLY
+     */
+    SAI_SEGMENTROUTE_ENTRY_ATTR_SUPPORTED_TLV_TYPE,
+
+    /**
+     * @brief List of TLVs for origination
+     *
+     * @type sai_tlv_list_t
+     * @flags CREATE_AND_SET
+     */
+    SAI_SEGMENTROUTE_ENTRY_ATTR_TLV_LIST,
+
+    /**
+     * @brief List of Segments to Originate
+     *
+     * @type sai_segment_list_t
+     * @flags MANDATORY_ON_CREATE | CREATE_AND_SET
```

Comment:
Sińce new 2 apibtypes are added parser.pl And saimrtadatsttpes attrvaluetypes needs to be updated aswell as saisanitycheck.c